bool push0
bool push1
bool dip0
bool dip1
bool dip2
bool dip3
bool up
bool down
bool left
bool right
bool center

